backend/pkg/api/connect/integration/transform_test.go

```diff
@@ -0,0 +1,293 @@
+package integration
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"testing"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	asrt "github.com/stretchr/testify/assert"
+	rqr "github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kadm"
+
+	"github.com/redpanda-data/console/backend/pkg/api"
+	"github.com/redpanda-data/console/backend/pkg/config"
+	v1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1"
+	v1alpha1connect "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1/dataplanev1alpha1connect"
+)
+
+var (
+	technicallyATransform = []byte{0x00, 0x61, 0x73, 0x6D}
+	transformTimeout      = 30 * time.Second
+)
+
+type response struct {
+	Transform *adminapi.TransformMetadata `json:"transform"`
+}
+
+func despawnTopic(ctx context.Context, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.DeleteTopic(ctx, connect.NewRequest(&v1alpha1.DeleteTopicRequest{
+		Name: topic,
+	}))
+	return err
+}
+
+func spawnTopic(ctx context.Context, partitionCount int32, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.CreateTopic(ctx, connect.NewRequest(&v1alpha1.CreateTopicRequest{
+		Topic: &v1alpha1.CreateTopicRequest_Topic{
+			Name:           topic,
+			PartitionCount: &partitionCount,
+			Configs: []*v1alpha1.CreateTopicRequest_Topic_Config{
+				{
+					Name:  "cleanup.policy",
+					Value: kadm.StringPtr("compact"),
+				},
+			},
+		},
+	}))
+	return err
+}
```

Comment:
Instead of using the connect client, could we use the kadm client like we did in other integration tests as well? If something breaks in the CreateTopic endpoint it would be cool that only the tests for this endpoint fail and not the entire integration testsuite because everyone uses this endpoint as well if that makes sense